addons/sale/models/sale_order.py

```diff
@@ -357,10 +357,23 @@ def _compute_require_payment(self):
         for order in self:
             order.require_payment = order.company_id.portal_confirmation_pay
 
-    @api.depends('require_payment')
+    @api.depends('require_payment', 'payment_term_id')
     def _compute_prepayment_percent(self):
         for order in self:
-            order.prepayment_percent = order.company_id.prepayment_percent
+            prepayment_percent = order.company_id.prepayment_percent or 1.0
+            payment_terms = []
+            if order.payment_term_id and order.payment_term_id.line_ids:
+                for line in order.payment_term_id.line_ids:
+                    payment_terms.append({
+                        'date': line._get_due_date(order.date_order or line.create_date),
+                        'percent': line.value_amount / 100,
```

Comment:
Tip: there is a `sorted` method on recordsets too.
Prefer manipulating recordsets when possible instead of aggregating side-data.

I believe something like this could work:
```python
payment_term_line = order.payment_term_id.line_ids.sorted(lambda line: line._get_due_date(order.date_order or line.create_date))[:1]
amount = payment_term_line.value_amount / 100
```